### PR TITLE
Add clarification on where to register entity plugins

### PR DESCRIPTION
**Pull request by ChickenF622**
Was running into the issue of running the draftail.registerPlugin within a DOMContentLoaded event listener and figured it would be a good note to add for anyone else that had this issue.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -    [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**:
    -   [x] **Please list which assistive technologies [^3] you tested**:
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: docs/extending/extending_draftail.md [hunk 1]
```diff
@@ -303,7 +303,7 @@ const Stock = (props) => {
 
 This is a straightforward React component. It does not use JSX since we do not want to have to use a build step for our JavaScript.
 
-Finally, we register the JS components of our plugin:
+Finally, we register the JS components of our plugin *note this is ran outside of an event listener such as DOMContentLoaded*:
```
Comment (lb-) [inline docs/extending/extending_draftail.md:306]: I'm concerned this line makes it more confusing, I understand you are talking about event listener in the context of the above code (I think). 

However, you advise that the registration should happen outside of an event listener and then suggest DOMContentLoaded which itself is an event. 

Are you advising that you need to register a plugin before content loaded or after? Or just saying it should be registered outside of any Callback?

Maybe a suggestion would be to add a comment line to the JS snippet below with where to put the code?

Sorry if I'm confused about the problem here but we just want to ensure we add clarity. 

As an aside; we don't use lots of bold/italics but rather admonitions if needed or just plain sentences. However, in this case I think a comment line in the JS snippet may be a good place for any clarification needed. 

Example
````
```javascript
// do not call this within any event listener, instead call as early as possible 
window.draftail.registerPlugin({
    type: 'STOCK',
```
````

Note. Not saying this text is right - just an example of where to put the comment.

**Review by lb- — CHANGES_REQUESTED** [review]
Thanks @ChickenF622 but I think this needs some work and maybe some more explanation on the PR description to help us understand what the problem here is exactly.

File: docs/extending/extending_draftail.md [hunk 2]
```diff
@@ -303,7 +303,7 @@ const Stock = (props) => {
...
 
 ```javascript
 window.draftail.registerPlugin({
```
Comment (thibaudcolas) [inline docs/extending/extending_draftail.md:309]:
```suggestion
// Register the plugin directly on script execution so the editor loads it when initialising.
window.draftail.registerPlugin({
```

File: docs/extending/extending_draftail.md (hunk 1 above)
Comment (thibaudcolas) [reply docs/extending/extending_draftail.md:306]:
```suggestion
Finally, we register the JS components of our plugin:
```

**Review by thibaudcolas — APPROVED** [review]
Thank you @ChickenF622, that’s an excellent clarification. I have rephrased according to LB’s points as I think this does read better as a code comment, and I agree mentioning an event name could be confusing.

**Comment by thibaudcolas** [comment]
@ChickenF622 congratulations on a first contribution! I’ve credited you using your GitHub handle `ChickenF622`, please let me know if you’d like to be credited as another name.

**Comment by ChickenF622** [comment]
Mark McOsker would be my preferred name for the credits.

**Comment by thibaudcolas** [comment]
Sweet! Updated :)

**Pull request merged**